### PR TITLE
CW-138 Docker Security

### PR DESCRIPTION
**Pull request by glebme**
# Change

_Ensure that docker containers and images are safer when they are spun up and deployed._

## Change reason

To ensure the security and integrity of our containers, docker security best practices were searched up and are being employed. The basic two things we can do to ensure safety when containers are spun up is to make sure that our files are read-only if they are accessed and  no new privileges able to be added if the server is breached. 

## Comments
- @gawdn anything else to add to how containers are spun up. Or anything else to address when I move on to sanitising what goes into an image? 
- Also would you know how to set 'least privileged users' from our wheatly server seeing that its not on our local machines and from reading it it seems can be done easily on local machine. Basically don't run the image as root, if thats even a concern on the server.

### CODE REVIEW

File: docker-compose.prod.yml [hunk 1]
```diff
@@ -3,8 +3,12 @@ services:
   production:
     image: production
     build: .
+    restart: always
     ports: 
       - '1323:1323'
+    security_opt:
+      - no-new-privileges
+    read_only: true
```
Comment (gawdn) [inline docker-compose.prod.yml:11]: Verify it still works when using an `ro` volume - oftentimes programs will like to write temp files

File: docker-compose.prod.yml [hunk 2]
```diff
@@ -3,8 +3,12 @@ services:
   production:
```
Comment (gawdn) [inline docker-compose.prod.yml:3]: Note that strictly the `docker-compose` isn't used in production but you can use it to validate the environment before deployment to wheatley (we use Kubernetes to manage the cluster)

**Review by gawdn — COMMENTED** [review]
@glebme Looks good - I wouldn't get too hung up on hardening the image too much, keeping it simple and using official images is normally enough. I handle security from the other perspective of controlling the Docker daemon and isolation from the host so you don't really need to worry much about that either.

Really the only things you will need to address in terms of image security is making sure there are no secrets leaking in the image. https://github.com/wagoodman/dive is a great tool to inspect the images yourself - you can see all the layers that are built up in the image.

File: docker-compose.prod.yml (hunk 2 above)
Comment (glebme) [reply docker-compose.prod.yml:3]: Ah. Good to know. Looking into the deployment environment is what I'm doing next. Thanks for the heads up.

File: docker-compose.prod.yml (hunk 1 above)
Comment (glebme) [reply docker-compose.prod.yml:11]: Will do.

Comment (glebme) [reply docker-compose.prod.yml:11]: Yep it is ro.

**Pull request merged**